### ISSUE TITLE
Dot bim to ifc 

### ISSUE THREAD

**Issue by Isaina** [RyugaRyuzaki/DotBimConvert#1]
Is possible to convert .bim to ifc just maybe wall, doors, openings, 

best regards

**Comment by RyugaRyuzaki** [comment]
yes, i think is possible . But i'm limited knowledge native ifc

**Comment by Isaina** [comment]
Here is python version https://github.com/Moult/dotbim-ifc

**Comment by RyugaRyuzaki** [comment]
no, i means this https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/annex_e/basic-geometric-shape/extruded-solid.html. I'm limited

**Comment by Isaina** [comment]
https://academy.ifcopenshell.org/posts/creating-a-simple-wall-with-property-set-and-quantity-information/

**Comment by RyugaRyuzaki** [comment]
Well, let me try with typescript, not python. But you can try with python

**Comment by Isaina** [comment]
https://github.com/IfcOpenShell/wasm-preview/tree/master

https://www.youtube.com/watch?v=MWJqGH7TOY8

**Comment by RyugaRyuzaki** [comment]
> https://github.com/IfcOpenShell/wasm-preview/tree/master
> 
> https://www.youtube.com/watch?v=MWJqGH7TOY8

yes, i tried it, but very slow to load wasm

**Comment by Isaina** [comment]
For me problems is that I don’t know dotbim , basically in dot bim there is geometry but how I know that this is a wall or windows , when converting from ifc if there is a  possibility to add meta data in description that this is ifcwall for egg would be helpfull imho

**Comment by RyugaRyuzaki** [comment]
> For me problems is that I don’t know dotbim , basically in dot bim there is geometry but how I know that this is a wall or windows , when converting from ifc there if there is a possibility to add meta data in description that this is ifcwall for egg would be helpfull imho

Here, https://github.com/RyugaRyuzaki/DotBimConvert/blob/main/Ifc/output/01/01-IfcWallStandardCase.bim#L3770C16-L3780 .The metadata

**Comment by Isaina** [comment]
https://github.com/andrewisen/bim-whale/tree/main

**Comment by Isaina** [comment]
Is maybe possible when converting put all ifc file in dotbim meta data , and have all in dotbim when converting back just remove dotbim part and write ifc , main problem is when dotbim geometry is changed but this can be checked when converting like wall dimension, windows dimensions, etc

**Comment by RyugaRyuzaki** [comment]
> Is maybe possible when converting put all ifc file in dotbim meta data , and have all in dotbim when converting back just remove dotbim part and write ifc , main problem is when dotbim geometry is changed but this can be checked when converting like wall dimension, windows dimensions, etc

Well, this is big issue, may be we can talk with Author to change it to version 2.Becuase i told him that .bim can use for sql database. May be we can discuss with author, what you think?

**Comment by Isaina** [comment]
For me this would be the easiest way to have ifc <-> .bim interoperability , to write into .bim and ifc file and when converting back check geometry , files would be larger but would be easy to handle.

**Comment by RyugaRyuzaki** [comment]
> For me this would be the easiest way to have ifc <-> .bim interoperability , to write into .bim and ifc file and when converting back check geometry , files would be larger but would be easy to handle.

Yes, you can tell the Author

**Comment by Isaina** [comment]
Basically I search for one dotbim to ifc converter , because on iPad Pro and iPhone pro there is a LiDAR room plan api, which gives usdz export, there is a tool mesh2mesh which can convert usdz to dotbim, and searching way to easy convert dotbim to ifc

**Comment by paireks** [comment]
.bim -> .ifc even though it exists (https://github.com/Moult/dotbim-ifc <- it was already mentioned above :)) it will never be perfect. It's because:
- geometry: you can translate any type of geometry to mesh, but not other way around. So once you have a .bim file you cannot "go back" to original types of geomtries such as extrusions, spheres, other solids, BREPs and so on :)
- data: you can translate any custom property sets to a dictionary, but not other way around. So once you have a .bim file you have dictionaries that cannot be easily translated into .ifc schema. There could be some additional rules hardcoded, such as e.g. changing every type Wall into IfcWall, Window into IfcWindow, and so on, but still it would be difficult to mantain precisely

That's why .bim -> .ifc mentioned can't create high-quality IFC file. It can satisfy some people, who use IFC as a 3d exchange file format, and don't really care about the schema, but then: why to use IFC? :)

**Comment by Isaina** [comment]
Here is ifc ts help https://github.com/hypar-io/IFC-gen

**Comment by Isaina** [comment]
here is clay work in progress so there is a easy way to generate ifc file without knowing ifc 

https://github.com/IFCjs/clay/commit/bcd1c227cf2a62eb4a460372285f7483a9de4828

Generally my idea is to go thru .bim file if is there is a section wall read dimension of stl object and convert to ifcwall, also for doors and all known elements if one element is not recognize leave it in stl format and put into ifc as is IFC4 version support this. But when clay would be ready for now it support only walls
